routing/router.go

```diff
@@ -1694,9 +1704,67 @@ func (r *ChannelRouter) sendToSwitch(route *route.Route, paymentHash [32]byte) (
 	firstHop := lnwire.NewShortChanIDFromInt(
 		route.Hops[0].ChannelID,
 	)
-	return r.cfg.SendToSwitch(
-		firstHop, htlcAdd, circuit,
+
+	// Using the created circuit, initialize the error
+	// decrypter so we can parse+decode any failures
+	// incurred by this payment within the switch.
+	errorDecryptor := &htlcswitch.SphinxErrorDecrypter{
```

Comment:
Can be moved down to where it is used.

Reply:
Not sure what you mean?

Reply:
Could be instantiated right above `GetPaymentResult` where it is used. To prevent unnecessary execution if it doesnt get there

Reply:
gotcha, fixed here: https://github.com/lightningnetwork/lnd/pull/3063/commits/0a2ff1cb8da7a0e89ca24c8e2efa94e9fa42552f